dotm_search.py

```diff
@@ -5,12 +5,54 @@
 Given a directory path, search all files in the path for a given text string
 within the 'word/document.xml' section of a MSWord .dotm file.
 """
-__author__ = "???"
+__author__ = "troyerjl"
 
 
-def main():
-    raise NotImplementedError("Your awesome code begins here!")
+import zipfile
+import argparse
+import os
+import sys
 
 
+# def open_function(args):
+#     full_path = args
+#     with ZipFile(full_path) as z:
+#         with z.open('word/document.xml') as dotm_doc:
+#             print(dotm_doc.read())
+
+
+def create_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--dir", default=".", help="read .dotm files from a directory")
+    parser.add_argument("text")
+    return parser
+
+
+def main(args):
+    parser = create_parser()
+    args = parser.parse_args()
+    
+    file_list = os.listdir(args.dir)
+    match_count = 0
+    search_count = 0
+
+    for file in file_list:
+        if not file.endswith(".dotm"):
+            print("Disregarding file {}".format(file))
```

Comment:
At least in ours, this wasn't part of the instructions.  Extra prints like that can clutter up your screen - use judiciously :)